### DISCUSSION TITLE
Porting finn examples to ZC702 Board

### DISCUSSION THREAD

**Discussion by ombhilare999** [Xilinx/finn#463] [Q&A]
I want to run a few CNN models on the ZC702 zynq Board, I have already ported the latest pynq image for zynq board.
Are there any blogs for porting finn examples to other boards?

Pynq Image for `ZC702`: https://github.com/DeepYNet/zynq-zc702-pynq
Board Link: https://www.xilinx.com/products/boards-and-kits/ek-z7-zc702-g.html

**Comment by maltanar** [comment]
For any of the `finn-examples` networks that already have a ZCU104 port, this should be quite straightforward. FINN already has a board definition for ZCU102, so you should simply be able to add `"ZCU102"` to the list of boards to build for, for example this line for the BNN-PYNQ examples:

https://github.com/Xilinx/finn-examples/blob/main/build/bnn-pynq/build.py#L20

Note that the MobileNet-v1 example already has a ZCU102 build target included:

https://github.com/Xilinx/finn-examples/blob/main/build/mobilenet-v1/build.py#L47

**Reply by ombhilare999** [reply]
I'm actually using ZC702 board not ZCU702.

**Reply by ombhilare999** [reply]
I see that supported boards have some folding config file here: 
https://github.com/Xilinx/finn-examples/tree/main/build/mobilenet-v1/folding_config
How can one generate this file for other xilinx boards?

**Reply by fpjentzsch** [reply]
The folding config itself is board-agnostic, but it determines how many resources (LUT, BRAM, URAM) the accelerator will require. If your device is significantly smaller than the given config for ZCU104 for example, you could start with a copy of that config and lower the PE & SIMD folding factors across all layers to decrease parallelism. Alternatively, you could run the build flow with a set fps target to let FINN determine a folding automatically (and then fine-tune from there).

You should be able to extend support to a new board by adding it to a few files. These 3 come to mind:
https://github.com/Xilinx/finn-base/blob/dev/src/finn/util/basic.py
https://github.com/Xilinx/finn-base/blob/dev/src/finn/util/platforms.py
https://github.com/Xilinx/finn/blob/dev/src/finn/transformation/fpgadataflow/templates.py

**Reply by rpitonak** [reply]
This issue https://github.com/Xilinx/finn/issues/368 may be helpful as well. It describes how to extend support for one of the end2end examples to a new board.
